Cargo.toml

```diff
@@ -12,37 +13,37 @@ description = "A tool for obtaining information from a TOML file on the command
 
 [dependencies]
 toml = "0.8"
-clap = { version = "4.5", features = ["derive", "usage", "help"] }
+clap = { version = "4.5", features = ["derive", "usage", "help"], optional = true}
 thiserror = "1.0.61"
-colored = { version = "2.1.0", optional = true }
-serde_json = { version = "1.0.120", features = [
+console = { version = "0.15.8", features = ["windows-console-colors"], optional = true }
+anyhow = { version = "1.0.86", features = ["backtrace"], optional = true }
+bat = { version = "0.24.0", optional = true, default-features = false, features = ["build-assets", "regex-fancy"] }
+
+[dependencies.serde_json] 
+version = "1.0.120"
+features = [
     "indexmap",
     "preserve_order",
     "raw_value",
     "unbounded_depth",
-], optional = true }
+]
+optional = true
 
 [features]
 default = ["json"]
-color = ["colored"]
 json = ["dep:serde_json"]
-
+bin = ["dep:clap", "dep:anyhow", "dep:bat", "dep:console"]
```

Comment:
Good catch on unbundling `clap` from the library. I usually start with keeping it bundled so there's an easy enough way to interface with the code "as if" it were the CLI but from other programs, but I think in this case that is 100% unnecessary

anyhow is a decent dependency here, and in general a better error handling process, but I'm not sure we should be mandating that the CLI come bundled with `bat` and `console`. Can you make these dependencies be optional for the CLI as well?

My current main hunch is that this is used more by CI systems than by people directly calling it, so visual features are secondary in my mind to providing something with minimal size + vulnerability surface. The less dependencies we require people to include, the less they need to update due to upstream issues